docs/DevGuide/BuildSystem.md

```diff
@@ -69,15 +69,33 @@ add_spectre_library(${LIBRARY} ${LIBRARY_SOURCES})
 
 target_link_libraries(
   ${LIBRARY}
-  <list_of_interfaces>
+  PUBLIC
+  <list_of_public_libraries>
+  PRIVATE
+  <list_of_private_libraries>
+  INTERFACE
+  <list_of_interface_libraries>
   )
 ```
-where `<list_of_interfaces>` is an alphabetized list of lines of the form
+where each `<list_of_X_libraries>` is an alphabetized list
+of libraries of the form
 ```
-  INTERFACE SOME_LIBRARY
+  SOME_LIBRARY
+  SOME_OTHER_LIBRARY
+  YET_ANOTHER_LIBRARY
 ```
-where `SOME_LIBRARY` is a library that must be linked in order for the
-new library to be successfully linked in an executable.
+The libraries listed under `INTERFACE` are those included in at
+least one `.hpp` file in `LIBRARY` but never used in any `.cpp` files
+in `LIBRARY`.  The libraries listed under `PRIVATE` are used in
+at least one `.cpp` file in `LIBRARY` but not in any `.hpp` file
+in `LIBRARY`.  The libraries listed under `PUBLIC` are used in at
+least one `.hpp` file and at least one `.cpp` file in `LIBRARY`.
+Note that a library counts as being "used in a .cpp file" if the
```

Comment:
In the text above this point things like `.cpp` and `PRIVATE` are in tics.  Make this part consistent with that.